server/build/webpack.js

```diff
@@ -148,5 +148,13 @@ export default async function createCompiler (dir, { hotReload = false } = {}) {
     customInterpolateName: function (url, name, opts) {
       return interpolateNames.get(this.resourcePath) || url
     }
-  })
+  }
+  try {
+    require(join(dir, 'webpack.js'))(config, hotReload)
```

Comment:
how about expecting the function to return the new config instead of mutating an argument? then it's more sensible to implement via `Object.assign`.

```js
// right here
config = require("...");
```

```js
// webpack.js in my project
module.exports = (config) => Object.assign(config, {
  // my custom config
})
```